starfish/core/spots/_detect_pixels/combine_adjacent_features.py

```diff
@@ -268,7 +268,17 @@ def _single_spot_attributes(
             }
 
         # we're back to 3d or fake-3d here
-        target_index = decoded_image[spot_attrs['z'], spot_attrs['y'], spot_attrs['x']]
+        bbox = spot_property.bbox
+        if len(bbox) == 6:
+            # 3d bbox
+            bb_non_zero = np.trim_zeros(np.ndarray.flatten(
+                decoded_image[bbox[0]:bbox[3], bbox[1]:bbox[4], bbox[2]:bbox[5]]))
+        else:
+            # flatten and remove zeros
+            bb_non_zero = np.trim_zeros(np.ndarray.flatten(
+                decoded_image[0, bbox[0]:bbox[2], bbox[1]:bbox[3]]))
+        # get the most repeated value
+        target_index = np.argmax(np.bincount(bb_non_zero))
```

Comment:
this may be returning 0, which then gets resolved to a 'nan' in the MERFISH unit test that i pointed out below. one way to quickly check for this is to find the non-zero most repeated value.  then re-run the merfish test and verify that the 'nan's go away.

Reply:
oh wait nm, np.trim_zeros gets rid of the 0s. hmmm. i'm not sure why you're still getting 'nans' in the merfish test case.

Reply:
oh wait, double nm. np.trim_zeros only removes leading and lagging 0s. 

so ```np.trim_zeros([0, 0, 1, 1, 0 0]) = [1, 1]```

but ```np.trim_zeros([1, 1,, 0, 0, 0, 0, 1]) = [1, 1,, 0, 0, 0, 0, 1]```

so you may not be removing 0s before finding the most repeated element. one way to fix this is like this:

```
target_candidates = np.ndarray.flatten( decoded_image[0, bbox[0]:bbox[2], bbox[1]:bbox[3]]))
non_zero_target_candidates = target_candidates[target_candidates != 0]
target_index = np.argmax(np.bincount(non_zero_target_candidates)))
```